File: Backend/BibliotecaAPI/BibliotecaAPI/Dtos/ResponseHistorialBibliotecaDto.cs
namespace BibliotecaAPI.Dtos
{
    public class ResponseHistorialBibliotecaDto
    {
        public int PrestamoId { get; set; }

        public int CantidadPrestada { set; get; }

        public DateOnly FechaPrestamo { get; set; }

        public bool Estado { get; set; }

        public DateOnly? FechaDevolucion { get; set; }

        public string NombreUsuario { get; set; } = null!;

        public string CedulaUsuario { get; set; } = null!;

        public string RolUsuario { get; set; } = null!;

        public string TituloMaterial { get; set; } = null!;

        public string NroIdentificacionMaterial { get; set; } = null!;

        public string TipoMaterial { get; set; } = null!;
    }
}

File: Frontend/BibliotecaIU/src/app/Biblioteca/Models/ResponseHistorialBibliotecaDto.ts
export interface ResponseHistorialBibliotecaDto {
  prestamoId: number;
  cantidadPrestada: number;
  fechaPrestamo: string; 
  estado: boolean;
  fechaDevolucion?: string; 

  nombreUsuario: string;
  cedulaUsuario: string;
  rolUsuario: string;

  tituloMaterial: string;
  nroIdentificacionMaterial: string;
  tipoMaterial: string;
}
